polaris/benchmark/_base.py

```diff
@@ -460,31 +453,16 @@ def evaluate(
         """
 
         # Instead of having the user pass the ground truth, we extract it from the benchmark spec ourselves.
-        # The `evaluate_benchmark` function expects the benchmark labels to be of a certain structure which
-        # depends on the number of tasks and test sets defined for the benchmark. Below, we build the structure
-        # of the benchmark labels based on the aforementioned factors.
-        test = self._get_test_set(hide_targets=False)
-        if isinstance(test, dict):
-            #
-            # For multi-set benchmarks
-            y_true = {}
-            for test_set_name, values in test.items():
-                y_true[test_set_name] = {}
-                if isinstance(values.targets, dict):
-                    #
-                    # For multi-task, multi-set benchmarks
-                    for task_name, values in values.targets.items():
-                        y_true[test_set_name][task_name] = values
-                else:
-                    #
-                    # For single task, multi-set benchmarks
-                    y_true[test_set_name][self.target_cols[0]] = values.targets
+        y_true_subset = self._get_test_set(hide_targets=False)
+
+        if isinstance(y_true_subset, dict):
+            y_true_values = {k: v.targets for k, v in y_true_subset.items()}
         else:
-            #
-            # For single set benchmarks (single and multiple task)
-            y_true = test.targets
+            y_true_values = y_true_subset.targets
```

Comment:
I did that in `evaluate_benchmark`, thinking it made sense to keep all the data manipulation parts together, since `evaluate_benchmark` is already very liberal with what it accepts. But if you think it makes more sense to do that here I'm happy to move it.

Reply:
Makes sense! I see it now. Thanks!